pkg/operatorsource/kubeclient.go

```diff
@@ -0,0 +1,54 @@
+package operatorsource
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+)
+
+// Kubeclient interface wraps the package level sdk.Create and sdk.Get methods provided by operator-sdk
+//
+// Why is this wrapper interface necessary?
+// Because these package level functions dont provide a way to inject a fake client,
+// it's hard to unit test reconciliation logic.
+// As a workaround, the reconciliation logic, instead of using the package level function directly will use
+// this interface. We can mock this interface and thus write proper unit test(s) for reconciliation logic.
+//
+// operator-sdk team is working on integrating controller-runtime
+// which will make it possible to inject fake clients in the future.
```

Comment:
Please ensure that we won't need to do a huge rewrite once operator-sdk adds this feature.

Reply:
I don't expect it to, it's a wrapper with a `decorator` pattern - it has the same exact signature as `sdk` functions. It should ensure the least amount of refactor.